caddy.go

```diff
@@ -867,3 +867,19 @@ var (
 	// by default if no other file is specified.
 	DefaultConfigFile = "Caddyfile"
 )
+
+// contextKey is a value for use with context.WithValue. Inspired by the
+// net/http source.
+type contextKey struct {
+	name string
+}
```

Comment:
@ericdreeves Ah, I didn't think about that. Yes, we can probably put `CtxKey` in the `caddy` package. Good idea. I don't see any negative consequences of that.